dashboard/modules/reporter/reporter_agent.py

```diff
@@ -154,7 +154,30 @@ def jsonify_asdict(o) -> str:
         "raylet_cpu", "CPU usage of the raylet on a node.", "percentage", ["ip", "pid"]
     ),
     "raylet_mem": Gauge(
-        "raylet_mem", "Memory usage of the raylet on a node", "mb", ["ip", "pid"]
+        "raylet_mem",
+        "Memory usage of the raylet on a node. Type includes uss and the remaining rss "
+        "on Linux, and just rss on other platforms.",
+        "MB",
+        ["ip", "pid", "mem_type"],
+    ),
+    "workers_count": Gauge(
+        "workers_count",
+        "Number of workers on a node.",
+        "count",
+        ["ip"],
+    ),
+    "workers_cpu": Gauge(
+        "workers_cpu",
+        "Total CPU usage of all workers on a node.",
+        "percentage",
+        ["ip"],
+    ),
+    "workers_mem": Gauge(
```

Comment:
Should we consider reporting these individually? It should be possible to aggregate them later on and I'm concerned we'll lose visibility this way. Do you have a sense for how much overhead it would add?

Reply:
@rkooo567 mentioned that Ray cluster can churn through many workers, so per-process export and aggregation is infeasible. One idea is to set up a topk aggregation. Maybe we can add top CPU per-process later?

Reply:
Hmm okay, I think the aggregate is fine for now, although I'm a bit concerned it's not enough information. If we have some idle workers or I/O-only workers, for example, it will be hard to see what the actual per-task memory usage is.

Could we try measuring the overhead of the per-worker metrics reporting? We can try a simple microbenchmark that churns through workers using `@ray.remote(max_calls=1)`.

Reply:
I agree it will be very nice to be able to look at a stacked plot of CPU usages for different workers across time on a node. I think the biggest cost for per-worker reporting is on prometheus, e.g. https://stackoverflow.com/questions/46373442/how-dangerous-are-high-cardinality-labels-in-prometheus. Maybe 1000 nodes * xxx workers would result in too many time series streams. In future depending on if the use case is timeline or top-k analysis, we can should be able to have a specialized but robust solution.